src/pydiverse/pipedag/context/run_context.py

```diff
@@ -650,7 +650,7 @@ def remove_names(self, stage: Stage, tables: list[Table], blobs: list[Blob]):
         self._request(
             "remove_names",
             stage.id,
-            [t.name for t in tables],
+            [t.get_run_context_name() for t in tables],
```

Comment:
It does not matter if we reference an external table multiple times within a stage:
```suggestion
            # ExternalTableReferences don't create a table in schema
            [t.name for t in tables if t.external_schema is None],
```